### PR TITLE
Wait for event method

### PR DESCRIPTION
**Pull request by KnaanBrazCanditech**
### Summary

*This pull request addresses the timeout issue in the getDataLayer function. Instead of having a random timeout for getDataLayer, we now have only one accessible function in gtmUtils called waitForEvent.*

### CODE REVIEW

File: utils/Gtm/gtmUtils.ts [hunk 1]
```diff
@@ -18,13 +14,28 @@ export const getGtmEvents = async (driver: IBaseDriver, eventName: EventName): P
...
   return events?.length || 0;
 };
 
-export const hasOneEventInstance = async (driver: IBaseDriver, eventName: EventName): Promise<boolean> => {
+const hasOneEventInstance = async (driver: IBaseDriver, eventName: EventName): Promise<boolean> => {
   const events = await getNumberOfGtmEvents(driver, eventName);
   return events === 1;
 };
 
+export const waitForEvent = async (driver: IBaseDriver, eventName: EventName, maxWaitingTime: number = 2_000): Promise<boolean> => {
```
Comment (eyalCanditech) [inline utils/Gtm/gtmUtils.ts:27]: I love it when the name has the unit of measure, in this case milli I assume
maxWaitingTimeMilli

**Comment by KnaanBrazCanditech** [comment]
> Is this the only solution available to us? It is a bit hacky

I guess 99% you won't need to wait more than 100 ms. 
But I didn't want to add just a random timeout. Also can solve future problems with event timing.

File: utils/Gtm/gtmUtils.ts [hunk 2]
```diff
@@ -18,13 +14,28 @@ export const getGtmEvents = async (driver: IBaseDriver, eventName: EventName): P
...
+  return new Promise((resolve) => {
+    const interval = setInterval(async () => {
+      const gotEvent = await hasOneEventInstance(driver, eventName);
```
Comment (eyalCanditech) [inline utils/Gtm/gtmUtils.ts:30]: what if you have more than one event?

**Review by eyalCanditech — APPROVED** [review]

File: utils/Gtm/gtmUtils.ts (hunk 2 above)
Comment (KnaanBrazCanditech) [reply utils/Gtm/gtmUtils.ts:30]: Good point. I will think about it and fix it in the next one

**Pull request merged**